### PR TITLE
refactor TurbomindModelConfig

### PR DESCRIPTION
**Pull request by lvhan028**

### CODE REVIEW

File: lmdeploy/messages.py [hunk 1]
```diff
@@ -123,18 +123,21 @@ class TurbomindEngineConfig:
...
+            `hf` meaning huggingface model(.bin, .safetensors), `meta_llama` being meta llama's format(.pth),
+            `awq` and `gptq` meaning the quantized model by AWQ and GPTQ, respectively.
+            If it is not specified, i.e. None, it will be extracted from the input model
         tp (int): the number of GPU cards used in tensor parallelism, default to 1
         session_len (int): the max session length of a sequence, default to None
         max_batch_size (int): the max batch size during inference, default to 128
         cache_max_entry_count (float): the percentage of gpu memory occupied by the k/v cache.
             For versions of lmdeploy between `v0.2.0` and `v0.2.1`, it defaults to 0.5, depicting the percentage of TOTAL GPU memory to be allocated to the k/v cache.
             For lmdeploy versions greater than `v0.2.1`, it defaults to 0.8, signifying the percentage of FREE GPU memory to be reserved for the k/v cache
+        cache_block_size (int): The policy to apply for KV block from the block manager, default to -1.
```
Comment (irexyc) [inline lmdeploy/messages.py:136]: cache_chunk_size?

File: lmdeploy/turbomind/deploy/converter.py [hunk 2]
```diff
@@ -228,6 +227,9 @@ def get_tm_model(model_path,
         else:
             assert 0, f'unsupported quant_config: {quant_config}'
 
+        engine_config.model_format = quant_method
+        group_size = _group_size
+
```
Comment (irexyc) [inline lmdeploy/turbomind/deploy/converter.py:232]: Shoud we assert _group_size != 0?

**Comment by irexyc** [comment]
test /mnt/140/InternLM/internlm2_5-7b-chat convert, load from workspace and load from hf.

File: lmdeploy/turbomind/deploy/converter.py (hunk 2 above)
Comment (lvhan028) [reply lmdeploy/turbomind/deploy/converter.py:232]: Yes. 
```
# Compatible to awq models that are quantized by lmdeploy (<=v0.3.0)
    if not group_size:
        group_size = 128

    if engine_config.model_format in ['awq', 'gptq']:
        assert group_size == 128, \
            f'model format is "{engine_config.model_format}" ' \
            f'but group_size is {group_size}. Currently, only 128 ' \
            'is supported'
```

**Review by lzhangzz — APPROVED** [review]

**Review by irexyc — APPROVED** [review]

**Pull request merged**